scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -235,6 +238,11 @@ def __init__(self, path: Optional[str], fname2path: dict[str, str],
         # before initializing those.
         self.raw: dict = self._merge_includes(raw, self.path)
 
+        if path is not None:
+            type_match = re.search(r'dts/bindings/(\w+)',
```

Comment:
I don't think `dts/bindings` should be hard coded, and you need to account for the fact that additional bindings folders might exist. I'm not sure we can have an util function in edtlib given that determining the type is basically something that tmwe can only do for upstream "official" Zephyr bindings that follow our own convention of organizing things in subfolders. There is nothing that forces out of tree bindings to do the same